src/rpcblockchain.cpp

```diff
@@ -2242,6 +1897,28 @@ UniValue projects(const UniValue& params, bool fHelp)
         entry.pushKV("Project", sProjectName);
         entry.pushKV("URL", sProjectURL);
 
+        if (mvCPIDs.size() > 0)
```

Comment:
Some small changes:
- Change size check to `mvCPIDs.empty()`. Same when doing `HarvestCPIDs`.
- Do a find (like [this](https://github.com/gridcoin-community/Gridcoin-Research/blob/master/src/appcache.cpp#L32)) on mvCPIDs or else we insert an element if it was missing before.
- Merge the `structcpid.initialized` check with the `IsResearcher` checks below it to remove one depth.